tests/functional/test_logger_powertools_formatter.py

```diff
@@ -288,3 +290,22 @@ def test_log_formatting(stdout, service_name):
 
     # THEN the formatting should be applied (NB. this is valid json, but hasn't be parsed)
     assert log_dict["message"] == '["foo bar 123 [1, None]", null]'
+
+
+def test_log_json_indent_compact_indent(stdout, service_name, monkeypatch):
+    # GIVEN a logger with default settings and WHEN POWERTOOLS_DEV is not set
+    if "POWERTOOLS_DEV" in os.environ:
+        monkeypatch.delenv(name="POWERTOOLS_DEV")
+    logger = Logger(service=service_name, stream=stdout)
+    logger.info("Test message")
+    # THEN the json should not be indented using constant.PRETTY_INDENT blank spaces
+    assert " " * constants.PRETTY_INDENT not in stdout.getvalue()
```

Comment:
```suggestion
    new_lines = stdout.getvalue().count("\n")
    assert new_lines == 1
```

Tip: non-indented log lines should have a single `\n`. While this assertion works for this test, it isn't always correct in the `test_log_json_pretty_indent` test because `assert " " * 0 in stdout.getvalue()` is also True.

Reply:
Hmm, at first I thought the same but then this test failed. For some reason,there was one new line (but I think my message at the time was two words, "Test message" and for some reason it got split in the middle.)

Reply:
also, should we not rather use `os.linesep`  instead of `\n`?

Reply:
I've commited your suggestion, but using `os.linesep`